src/System/Random/Internal.hs

```diff
@@ -499,9 +506,73 @@ class Uniform a where
   -- | Generates a value uniformly distributed over all possible values of that
   -- type.
   --
+  -- There is a default implementation via 'Generic':
+  --
+  -- >>> :set -XDeriveGeneric -XDeriveAnyClass
+  -- >>> import GHC.Generics (Generic)
+  -- >>> import System.Random.Stateful
+  -- >>> data MyBool = MyTrue | MyFalse deriving (Show, Generic, Finite, Uniform)
+  -- >>> data Action = Code MyBool | Eat (Maybe Bool) | Sleep deriving (Show, Generic, Finite, Uniform)
+  -- >>> gen <- newIOGenM (mkStdGen 42)
+  -- >>> uniformListM 10 gen :: IO [Action]
+  -- [Code MyTrue,Code MyTrue,Eat Nothing,Code MyFalse,Eat (Just False),Eat (Just True),Eat Nothing,Eat (Just False),Sleep,Code MyFalse]
+  --
   -- @since 1.2.0
   uniformM :: StatefulGen g m => g -> m a
 
+  default uniformM :: (StatefulGen g m, Generic a, GUniform (Rep a)) => g -> m a
+  uniformM = fmap to . (`runContT` pure) . guniformM
+
+class GUniform f where
```

Comment:
Use of ContT is nonobvious so it would be nice to add comment about it

```suggestion
-- | Default implementation of Uniform type class in terms
-- generics. Note use of ContT. Without it uses of fmap and bind are
-- fully polymorphic and GHC can't inline or specialize it. ContT
-- makes fmap and binds used in @gniformM@ monomorphic so GHC able to
-- specialize code instance reasonably close to handwritten one
class GUniform f where
```